### ISSUE TITLE
[BUG]: Operands could not be broadcast together with shapes (15831,12) (15831,15830,13)

### ISSUE THREAD

**Issue by LIZW2019** [perslab/CELLEX#34]
Dear friend,

I came across the Value Error when I used cellex on my data like this:

![image](https://user-images.githubusercontent.com/54759510/207790801-1b808d64-1db9-4dff-abf5-9e08b8689a6d.png)

And I fixed this bug by modifying the line 44  in cellex/metrics/det.py like this:
Change 
`n_cells = np.array([n_cells.values] * mean.shape[0]) # faster than count`

To
`n_cells = np.array(n_cells.values * mean.shape[0]) # faster than count`

My lab member use the same tools but he has never come across this problem. I am wondering if this bug is cause by the different version of numpy, because it seems to be an array calculation problem. My numpy version is 1.21.6, and my cellex version is 1.2.2.

I would be thankful If this problem can be fixed for broader environment toleration. And I would appreciate it if you can tell me the real reason for the different calculation rule in my computer.

Thanks a lot!

Amy

**Comment by LIZW2019** [comment]
The bug seems to disappear when I renew the environment...

**Comment by tstannius** [comment]
Dear Amy,

My apologies for the late reply - I unfortunately did not receive a notification - and thanks for submitting this issue and your debugging efforts!

This issue appears to be very similar to the one reported [here](https://github.com/perslab/CELLEX/issues/33).
I would suggest you try the same steps as I have suggested Andrew [here](https://github.com/perslab/CELLEX/issues/33#issuecomment-1377729364).
I.e.
1) Get a full stack-trace.
2) Create and test a CELLEX environment where the library versions are exactly the same as those listed in `requirements.txt`

Supposedly the reason for the different computation you observe is that the `n_cells` does not have the expected number of dimensions. This may be because numpy behaves differently with respect to how it interprets the line you have posted:
```python
n_cells = np.array([n_cells.values] * mean.shape[0])
```
VS
```python
n_cells = np.array(n_cells.values * mean.shape[0])
```
It's been some time since I wrote this - if I recall correctly, this line is supposed to result in a 2D array, but if n_cells.values is already 2D, it will become 3D, causing the error you observe further down.

With respect to fixing the problem, we are looking into a new, faster version of CELLEX, but likely not updating the existing version.
